packages/grafana-ui/src/components/Table/TableNG/utils.ts

```diff
@@ -71,58 +81,178 @@ export function shouldTextWrap(field: Field): boolean {
   return Boolean(cellOptions?.wrapText);
 }
 
-// matches characters which CSS
+/**
+ * @internal creates a typography context based on a font size and family. used to measure text
+ * and estimate size of text in cells.
+ */
+export function createTypographyContext(fontSize: number, fontFamily: string, letterSpacing = 0.15): TypographyCtx {
+  const font = `${fontSize}px ${fontFamily}`;
+  const canvas = document.createElement('canvas');
+  const ctx = canvas.getContext('2d')!;
+
+  ctx.letterSpacing = `${letterSpacing}px`;
+  ctx.font = font;
+  const txt =
+    "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s.";
+  const txtWidth = ctx.measureText(txt).width;
+  const avgCharWidth = txtWidth / txt.length + letterSpacing;
+  const { count } = varPreLine(ctx);
+
+  return {
+    count,
+    ctx,
+    font,
+    avgCharWidth,
+    estimateLines: getTextLineEstimator(avgCharWidth),
+    wrappedCount: count as LineCounter,
+  };
+}
+
+/**
+ * @internal returns a line counter which guesstimates a number of lines in a text cell based on the typography context's avgCharWidth.
+ */
+export function getTextLineEstimator(avgCharWidth: number): LineCounter {
+  return (value, width) => {
+    if (!value) {
+      return -1;
+    }
+
+    // we don't have string breaking enabled in the table,
+    // so an unbroken string is by definition a single line.
+    const strValue = String(value);
+    if (!spaceRegex.test(strValue)) {
+      return -1;
+    }
+
+    const charsPerLine = width / avgCharWidth;
+    return strValue.length / charsPerLine;
+  };
+}
+
+/**
+ * @internal return a text line counter for every field which has wrapHeaderText enabled.
+ */
+export function buildHeaderLineCounters(fields: Field[], typographyCtx: TypographyCtx): LineCounterEntry[] | undefined {
+  const wrappedColIdxs = fields.reduce((acc, field, idx) => {
+    if (field.config?.custom?.wrapHeaderText) {
+      acc.push(idx);
+    }
+    return acc;
+  }, [] as number[]);
```

Comment:
fix this as, and just introduce the wrap method early that we're gonna need in the max lines PR anyway